src/MaintenanceModeCommand.php

```diff
@@ -132,7 +132,26 @@ private function get_maintenance_mode_status() {
 
 		$maintenance_file = trailingslashit( $wp_filesystem->abspath() ) . '.maintenance';
 
-		return $wp_filesystem->exists( $maintenance_file );
+		if ( ! $wp_filesystem->exists( $maintenance_file ) ) {
+			return false;
+		}
+
+		// We use the timestamp defined in the .maintenance file
+		// to check if the maintenance is available.
+		$upgrading = 0;
+
+		$contents = $wp_filesystem->get_contents( $maintenance_file );
+		$matches  = [];
+		if ( preg_match( '/upgrading\s*=\s*(\d+)\s*;/i', $contents, $matches ) ) {
+			$upgrading = (int) $matches[1];
+		} else {
+			WP_CLI::warning( 'Unable to read the maintenance file timestamp, non-numeric value detected.' );
+		}
+
+		if ( ( time() - $upgrading ) >= 10 * MINUTE_IN_SECONDS ) {
```

Comment:
Can we include some mention that this logic is from `wp_is_maintenance_mode()`?

Reply:
Sure, done.